File: app/src/main/java/com/numerad/quickseries/dagger/AppModule.java
package com.numerad.quickseries.dagger;

import android.content.Context;

import javax.inject.Singleton;

import androidx.annotation.NonNull;
import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private Context context;

    public AppModule(@NonNull Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return context;
    }

}

File: app/src/main/java/com/ztd/interview_test/infrustructure/di/AppModule.kt
package com.ztd.interview_test.infrustructure.di

import android.app.Application
import android.content.Context
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.ztd.interview_test.infrustructure.data.AssetHelper
import com.ztd.interview_test.infrustructure.data.AssetHelperImp
import dagger.Module
import dagger.Provides
import io.reactivex.disposables.CompositeDisposable
import javax.inject.Singleton

/**
 * Created by Mahdi_ZareTahghighDoost(ZTD)
 *  on 1/10/2020.
 */
@Module
class AppModule {

    @Provides
    @Singleton
    internal fun provideCompositeDisposable(): CompositeDisposable {
        return CompositeDisposable()
    }

    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }

    @Provides
    @Singleton
    fun provideGson(): Gson {return GsonBuilder().excludeFieldsWithoutExposeAnnotation().create()}


    @Provides
    @Singleton
    fun provideAssetHelper(assetHelperImp: AssetHelperImp):AssetHelper = assetHelperImp


}